lib/Wrapper/OptSchedReg.h

```diff
@@ -0,0 +1,30 @@
+#ifndef OPT_SCHED_REG
+#define OPT_SCHED_REG
+
+#include "OptimizingScheduler.h"
+#include "llvm/CodeGen/MachineScheduler.h"
+
+using namespace llvm;
+
+namespace llvm {
+namespace opt_sched {
+
+// Create OptSched ScheduleDAG.
+static ScheduleDAGInstrs *createOptSched(MachineSchedContext *C) {
+  ScheduleDAGMILive *DAG =
+      new ScheduleDAGOptSched(C, std::make_unique<GenericScheduler>(C));
+  // DAG->addMutation(createCopyConstrainDAGMutation(DAG->TII, DAG->TRI));
+  // README: if you need the x86 mutations uncomment the next line.
+  // addMutation(createX86MacroFusionDAGMutation());
+  // You also need to add the next line somewhere above this function
+  //#include "../../../../../llvm/lib/Target/X86/X86MacroFusion.h"
```

Comment:
Don't give a relative path like that, especially not that many directories up. Ideally, this would be a build system option, probably pulling the header file in under OptSched and saying to include the OptSched-local file.

Reply:
This is actually coming from the current master branch -- just ported over to OptSchedReg. It's used to generate execution time results for X86 -- interesting idea to have that be a build option.

Reply:
Got it. I didn't realize the code was just moved. Don't worry about it then.